### ISSUE TITLE
Basic Auth problem

### ISSUE THREAD

**Issue by jmaitrehenry** [containous/traefik#3448]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Update from 1.5 to 1.6.2, the basic auth from my service doesn't work anymore.

### What did you expect to see?

Basic auth continue to work

### What did you see instead?

`HTTP/2 401`

### Output of `traefik version`: (_What version of Traefik are you using?_)
 
```
"Image": "traefik:1.6@sha256:31de7b1e00170aade28f808f879ee5b237ab85461e5bbd46a40f3d609252d5b4"
```

### More informations

I create a new token using the documentation:
```
echo $(htpasswd -nb user password) | sed -e s/\\$/\\$\\$/g
user:$$apr1$$GKODusU7$$TQked1tW25DhbihdfmrCY0

curl -v -u user:password https://myendpoint
> GET / HTTP/2
> Host: myendpoint
> Authorization: Basic dXNlcjpwYXNzd29yZA==
> User-Agent: curl/7.54.0
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 401
< content-type: text/plain
* Authentication problem. Ignoring this.
< www-authenticate: Basic realm="traefik"
< content-length: 17
< date: Wed, 06 Jun 2018 19:36:24 GMT
<
401 Unauthorized
```

Docker service configuration:
```
docker service inspect myservice
<zip>
"traefik.frontend.auth.basic": "user:$apr1$GKODusU7$TQked1tW25DhbihdfmrCY0"
<zip>
```

Side note, I remove the Authorization header between the FE and the BE:
`"traefik.frontend.headers.customRequestHeaders": "Authorization:-",`

**Comment by jbdoumenjou** [comment]
Hi @jmaitrehenry,

Thanks for your in interest in Træfik.

The order of the middleware was changed since the #3187 to fix a security issue.
The secured middleware will apply the redirection with needed headers before the authentication to avoid passing the user/password over http instead of https.
Now, the modification of the header is applied before the authentication, so it doesn’t work anymore.

**Comment by jmaitrehenry** [comment]
Hi @jbdoumenjou, If I understand, in my case, we don't need the redirection, the request is already made on the https endpoint.

But, as I ask to Traefik to not send the basic-auth header to the backend because it will conflict with the authentification of the application, the traefik basic-auth do not work anymore for my case, right?

**Comment by jbdoumenjou** [comment]
Indeed

**Comment by jbdoumenjou** [comment]
Perhaps, you could explain me your usecase, so I can try to help you.

**Comment by jmaitrehenry** [comment]
@jbdoumenjou The case I have, I have a software that have an old auth library that accept a basic auth header and the old cookie way for connecting a user. It's a service protected by an auth-basic in the LB side because I don't want to expose it to everyone as it's old and I don't know how secure this app is.

So, the auth-basic header should be removed from the LB because the application should not be aware of it, else it will try to connect you and failed.

**Comment by jmaitrehenry** [comment]
For now, I fixed that by adding a nginx proxy between Traefik and my service. Nginx configuration is as simple as that:
```
server {
  listen 8080;

  location / {
    proxy_set_header Authorization  "";
    proxy_pass http://myservice/;
  }
}
```

You can close it if you think you will not fix it as a workaround exist. It's not the best because you add an another layer, but it works.

**Comment by traefiker** [comment]
Closed by #3606.
